extensions/vertx-http/deployment-spi/src/main/java/io/quarkus/vertx/http/deployment/spi/NonApplicationRootPathBuildItem.java

```diff
@@ -0,0 +1,19 @@
+package io.quarkus.vertx.http.deployment.spi;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+
+/**
+ * Intended for extensions that may require to consume the non Application Root path, but do not have available the
+ * original NonApplicationRootPathBuildItem from the vertx-http extension.
+ */
+public final class NonApplicationRootPathBuildItem extends SimpleBuildItem {
```

Comment:
The question here is if we want to move the original `NonApplicationRootPathBuildItem` to the SPI module. I believe the original one has a lot of other stuff not required by other extensions, so maybe it makes sense to keep them separate.

Additional context: The motivation for this is that we don't want to force the `vertx-http` extension as a dependency.

Reply:
"I believe the original one has a lot of other stuff not required by other extensions, so maybe it makes sense to keep them separate."

That isn't accurate. Other extensions are currently using the NonApplicationRootPathBuildItem as a route builder: 
```
nonApplicationRootPathBuildItem.routeBuilder()
                .routeFunction(config.export.json.path, recorder.route())
                .handler(recorder.getHandler())
                .blockingRoute()
                .build()
```
This usage is very much based on having vertx-http present, so I don't think moving that to an SPI package is necessary (?!)

In this case, what Observability needs is just a way to find the configured paths (as resolved against each other). My worry is that extensions will use this to begin cobbling together their own routes, which is a path we'd rather avoid.